### PR TITLE
Introduce bun and cleanup deps

### PR DESCRIPTION
**Pull request by heikir**

### CODE REVIEW

File: .github/actions/env/action.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    required: true
+    default: "20.x"
+runs:
+  using: "composite"
+  steps:
+    - uses: actions/setup-node@v3
+      with:
+        node-version: ${{ inputs.node-version }}
+        registry-url: "https://registry.npmjs.org"
+    - uses: pnpm/action-setup@v2
```
Comment (christophwitzko) [inline .github/actions/env/action.yaml:15]: we could drop pnpm completely

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -12,24 +12,24 @@ jobs:
...
-        node-version: [18.x]
+        node-version: [20.x]
     steps:
       - uses: actions/checkout@v3
-      - uses: ./.github/actions/pnpm
+      - uses: ./.github/actions/env
         with:
           node-version: ${{ matrix.node-version }}
       - name: Test
         run: pnpm test
```
Comment (christophwitzko) [inline .github/workflows/ci.yml]: should be `bun run test`

**Review by christophwitzko — APPROVED** [review]
LGMT 👍

File: .github/workflows/ci.yml (hunk 2 above)
Comment (heikir) [reply .github/workflows/ci.yml]: Fixed. It's interesting to note `bun test` gives a different test suite result. For anyone who might be confused initially in the future.

File: .github/actions/env/action.yaml (hunk 1 above)
Comment (heikir) [reply .github/actions/env/action.yaml:15]: And run publish using npm?

Comment (christophwitzko) [reply .github/actions/env/action.yaml:15]: yes, use npm for that 👍

**Pull request merged**